numpy/core/src/common/npy_cpu_features.c

```diff
@@ -275,47 +302,77 @@ npy__cpu_try_disable_env(void)
             notsupp_cur[flen] = ' '; notsupp_cur += flen + 1;
             goto next;
         }
-        // Finally we can disable it
-        npy__cpu_have[feature_id] = 0;
+        // Finally we can disable or mark for enabling
+        npy__cpu_have[feature_id] = disable ? 0:2;
     next:
         feature = strtok(NULL, delim);
     }
+    if (!disable){
+        // Disables any unmarked dispatched feature.
+        #define NPY__CPU_DISABLE_DISPATCH_CB(FEATURE, DUMMY) \
+            if(npy__cpu_have[NPY_CAT(NPY_CPU_FEATURE_, FEATURE)] != 0)\
+            {npy__cpu_have[NPY_CAT(NPY_CPU_FEATURE_, FEATURE)]--;}\
+
+        NPY_WITH_CPU_DISPATCH_CALL(NPY__CPU_DISABLE_DISPATCH_CB, DUMMY) // extra arg for msvc
+    }
 
     *nexist_cur = '\0';
     if (nexist[0] != '\0') {
         *(nexist_cur-1) = '\0'; // trim the last space
-        if (PyErr_WarnFormat(PyExc_RuntimeWarning, 1,
+        if (disable &&
+            PyErr_WarnFormat(PyExc_ImportWarning, 1,
                 NPY__CPU_ENV_ERR_HEAD
                 "You cannot disable CPU features (%s), since "
                 "they are not part of the dispatched optimizations\n"
                 "(" NPY_WITH_CPU_DISPATCH ").",
-                nexist
+                action, nexist
         ) < 0) {
             return -1;
+        } else if (!disable)
+        {
+            PyErr_Format(PyExc_RuntimeError,
+                NPY__CPU_ENV_ERR_HEAD
+                "You cannot enable CPU features (%s), since "
+                "they are not part of the dispatched optimizations\n"
+                "(" NPY_WITH_CPU_DISPATCH ").",
+                action, nexist
+            );
+            return -1;
         }
+        return 0;
     }
 
     *notsupp_cur = '\0';
     if (notsupp[0] != '\0') {
         *(notsupp_cur-1) = '\0'; // trim the last space
-        if (PyErr_WarnFormat(PyExc_RuntimeWarning, 1,
+        if (disable && PyErr_WarnFormat(PyExc_ImportWarning, 1,
```

Comment:
```suggestion
        if (PyErr_WarnFormat(PyExc_ImportWarning, 1,
            "%sYou cannot %s CPU features (%s), since "
            "they are not supported by your machine.",
            err_head, act_name, notsupp
        ) < 0) {
            return -1;
        }
```
same here too

Reply:
Here enable/disable have different semantics:

- disabling an unsupported feature results in a warning
- enabling an unsupported feature results in an error